internal/controllers/nmc_reconciler.go

```diff
@@ -477,29 +519,28 @@ func (h *nmcReconcilerHelperImpl) SyncStatus(ctx context.Context, nmcObj *kmmv1b
 				continue
 			}
 
-			status.Config = &config
-			status.InProgress = false
+			if irsName, err := getImageRepoSecretName(&p); err != nil {
+				logger.Info(
+					utils.WarnString("Error while looking for the imageRepoSecret volume"),
+					"error",
+					err,
+				)
+			} else if irsName != "" {
+				status.ImageRepoSecret = &v1.LocalObjectReference{Name: irsName}
+			}
+
+			status.ServiceAccountName = p.Spec.ServiceAccountName
 
 			podLTT := GetContainerStatus(p.Status.ContainerStatuses, workerContainerName).
 				State.
 				Terminated.
 				FinishedAt
 
-			status.LastTransitionTime = &podLTT
-			podsToDelete = append(podsToDelete, p)
-		case v1.PodPending, v1.PodRunning:
-			status.InProgress = true
-			updateModuleStatus = true
-			// TODO: if the NMC's spec changed compared to the Pod's config, recreate the Pod
-		default:
-			errs = append(
-				errs,
-				fmt.Errorf("%s: unhandled Pod phase %q", podNSN, phase),
-			)
-		}
+			status.LastTransitionTime = podLTT
 
-		if updateModuleStatus {
 			nmc.SetModuleStatus(&nmcObj.Status.Modules, *status)
+
+			podsToDelete = append(podsToDelete, p)
```

Comment:
what about default switch case? just in case, at least we will have some log

Reply:
Do we want to log in that case? I'd rather avoid having messages that are not really useful.

Reply:
I think that in that case it means that something is really fucked up, and we would like to see notification in log. We are not going to see multiple message in usual flow

Reply:
One of the other phases is `Pending`, I don't think we want to see anything for it.

Reply:
ok, then lets just log the Unknown, just in case